Add About Landing page

pollygee opened the pull request:
@meg-gutshall @seanmarcia I've done as much as I know how to do on this page.  
I don't know how to get the link to the anchor work from the exclusivity and equity section.
I don't know how to make the background change color all the way across the page over the sections that don't span the whole page and I don't know the color for that background stripe.

I don't know if it's helpful, but I can try to keep building out as much as I know how to on some pages and have someone else do some of the styling?  If that's not helpful, let me know what would be helpful.

pollygee committed:
Add About Landing page

pollygee committed:
Fix spacing for core values

pollygee committed:
Fix some spacing

pollygee commented on assets/css/style.css:
```diff
@@ -4,6 +4,7 @@
   Version: 1.0
 */
```
I'm not sure why there are so many changes to this file.  I rebased from master and there were big changes in master.  I'm not sure why they're showing up as a diff from master to my branch though.  🤔

meg-gutshall replied:
Do you have your tab set to 4 spaces? I have mine set to 2. It looks like that could be what's causing this.

meg-gutshall replied:
Also there's no newline spacing between CSS rules on the master branch and there is on this one. Do you have a formatting plugin or something maybe that's running automatically on the CSS?

meg-gutshall commented:
Hey Polly! From what I've seen so far this looks good! I'll fix up the CSS and push back those changes later tonight with a description of what I did. There's some utility classes built in so it's easier to use. 😄 

Thank you for your work! Even just putting the basic build up there and letting me fix up the CSS is helpful—or else I'd just be building the whole page on my own. I will get back to you soon!

meg-gutshall committed:
Create mixins partials directory

meg-gutshall committed:
Add background color and styling to section_header partial

meg-gutshall committed:
Merge branch 'master' into about_landing

meg-gutshall committed:
Create about_gooders.yml file

Create .yml file to hold content for the 'What Is a Gooder?' block and add it to the newly created _data/about_us directory so it's easier to add/edit content.

meg-gutshall committed:
Add nested paragraphs and rename file

I created nested paragraphs to render the text with line breaks between paragraphs. Also add file to the newly created _data/about_us directory so it's easier to add/edit content.

meg-gutshall committed:
Rename first block in 'About Us' page

meg-gutshall committed:
Move values.yml to about_us data directory and update font awesome code

meg-gutshall committed:
Add .block-header class and change fonts

Increase .title-header font size and decrease .block-header font size

meg-gutshall committed:
Add font awesome class to styles

meg-gutshall committed:
Remove about floats, use bootstrap columns instead

meg-gutshall committed:
Make block_header more functional

Add newly created .block-header CSS class, create more space between title and description, attach a conditional to rendering the description

meg-gutshall committed:
Make section_header more functional

Create more space between title and description, attach a conditional to rendering the description, add padding to the bottom of the partial

meg-gutshall committed:
Add front matter to render section_header

Added title, description, bg, and style variables, which will be used in the mixins/section_header partial called in the body of the page. I changed the title from "Landing Page" to "About Us" as this is what is rendered on the site.

meg-gutshall committed:
Add block variables to front matter

Iterate over the blocks in the body of the page. The mixins/block_header partial will be rendered only if the block has a title variable. Below that, the content of the block will be rendered.

meg-gutshall committed:
Clean up bootstrap column size

meg-gutshall committed:
Make variable assignment for iteration simplicity

meg-gutshall committed:
Style changes

Add block background conditional statement, adjust top and bottom padding elements, adjust bootstrap column sizing

meg-gutshall committed:
Remove headers

meg-gutshall committed:
Adjust tabs

meg-gutshall committed:
Iterate through gooder data

meg-gutshall committed:
Add front matter and conditional statement

Wrap the mixins/block_header partial in a conditional statement to render only if the block has a page title. Add missing front matter and attributes to page.

meg-gutshall reviewed (approved):
Hey @pollygee! I made some changes so I can merge this into master. I'm going to leave this branch live so you have time to take a look at it. I made my commits pretty granular and added further description to some of them.

I'm also going to set up the pages we have left to build out with the front matter I _think_ we'll need for them but that could change. My goal behind building it out this way is to make it as easy as possible for Gia to add and edit content when she needs to.

meg-gutshall commented:
As far as your original questions go...

<br>

> I don't know how to get the link to the anchor work from the exclusivity and equity section.

Me neither! That is something I can look into or if you decide to tackle it, let me know! 🙂 

<br>

> I don't know how to make the background change color all the way across the page over the sections that don't span the whole page and I don't know the color for that background stripe.

So this will change depending on whether you're building the page with multiple "blocks" as I've started calling them in my head (partials), or one partial for the page content. You can see in this commit 6e15fc1f80b14eb7dd360380d0ba1dd962f71a58 that I added the conditional `{% if block.bg %}{{ block.bg }}{% endif %}` line in the `section` class. This draws from the `about-us/landing-page.md` front matter and will include the class that gets passed in, if one exists. For a page that uses only one partial to hold its content, you'll use `page.bg` instead of `block.bg`.
In both cases, the `mixins/section_header.html` partial draws its background color from the `page.bg` front matter attribute. If the page has blocks, the `mixins/block_header.html` partial gets the background color from `block.bg`.
There are `background-color` variables set in the `default.css` file, but we've mainly been sticking to `grey-bg` and `white-bg`. Also in that file are the margin and padding classes you can see used throughout the code.

<br>

> I don't know if it's helpful, but I can try to keep building out as much as I know how to on some pages and have someone else do some of the styling? If that's not helpful, let me know what would be helpful.

I know jumping into a project is rough, but I'm super grateful for your help! 😍  I'm going to create a new branch tonight or tomorrow called mvp and just rip out a lot of the stuff we don't need right now. Hopefully that will make it easier for you to navigate the project. Please let me know if there's anything else I can do that will help you get more familiar with the code!

pollygee commented:
Thanks @meg-gutshall for all the explanation.  I think I mostly understand what you said....though the new vocabulary is getting me a bit haha! But it's def all clearer than when I started.  I'll get it at some point.  thanks for fixing the rest of that up.  I felt bad it being in a partially completed state.  I'm glad it's better than nothing though!

I'm not confused with all the stuff that's in there, so don't rip anything out on my behalf.  I wonder if we could just comment pages out we're not going to show for MVP?  Whatever you think is best though 🤷
